### DISCUSSION TITLE
Windows平台如何解析Composition为VideoComposition类型的pag？

### DISCUSSION THREAD

**Discussion by scuhappy** [Tencent/libpag#1019] [Q&A] [answered]
目前解码BitMapComposition或者VectorComposition都没问题，但是解析VideoComposition失败。用了自带的qt的例子，解析VideoComposition的pag会报错，软解码器找不到。
我自己debug的时候发现，在Windows上找不到解码器，解码失败。看了下windows的NativePlatform的实现，没有实现解码器相关的函数。

**Comment by domchen** [comment, marked as answer]
在默认的Platform基类里有注册libavc的软件解码器，你检查一下是不是自己在CmakeLists编译参数里关闭了libavc模块，否则除了web平台，其他平台默认都会带这个。自带的qt demo我在Windows上运行是可以正常解码视频序列帧的。但是libavc性能比较差，由于开源协议不兼容，我们没法直接内置ffmpeg，如果你需要性能更好的软件解码器，可以参考这个项目https://github.com/libpag/ffavc 编译一下出动态库，然后自己注册给pag使用就行。ffavc仓库里有注册的demo可以参考一下

**Reply by scuhappy** [reply]
多谢，看了一下，libavc之前减包的时候被移除了。。。
